middleware/etcd/msg/service.go

```diff
@@ -21,6 +21,10 @@ type Service struct {
 	Mail     bool   `json:"mail,omitempty"` // Be an MX record. Priority becomes Preference.
 	TTL      uint32 `json:"ttl,omitempty"`
 
+	// Fqdn of the service.  For wildcard requests, we need to know the name
+	// of each service so we can put them in the answer.
+	Fqdn string `json:"fqdn,omitempty"`
+
```

Comment:
You can't really free-wheely add stuff do this `msg.Service` as it underpins quite a lot (like also SkyDNS). We also want to keep this struct smallesh - is there another solution?